overlord/ifacestate/ifacemgr.go

```diff
@@ -275,9 +275,23 @@ func (m *InterfaceManager) doRemoveSnapSecurity(task *state.Task, _ *tomb.Tomb)
 	if err != nil {
 		return err
 	}
-	// TODO: remove all connections from the state
-	if err := m.repo.RemoveSnap(snapName); err != nil {
 
+	conns, err := getConns(task.State())
```

Comment:
We cannot do this here. We need to be able to distinguish between removing the profiles, and discarding the connections. Otherwise we'll end up killing user data when deactivating a snap, or removing an old revision.

I suggest clarifying the security tasks we have, with the following set:
* setup-profiles
* remove-profiles
* discard-conns